src/main/java/io/mbab/sda/groupproject/menu/action/CreateSongAction.java

```diff
@@ -1,4 +1,78 @@
 package io.mbab.sda.groupproject.menu.action;
 
-public class CreateSongAction {
+import io.mbab.sda.groupproject.entity.Album;
+import io.mbab.sda.groupproject.entity.Song;
+import io.mbab.sda.groupproject.menu.CustomScanner;
+import io.mbab.sda.groupproject.menu.MenuActionContext;
+import io.mbab.sda.groupproject.repository.AlbumRepository;
+import io.mbab.sda.groupproject.repository.SongRepository;
+import lombok.RequiredArgsConstructor;
+
+@RequiredArgsConstructor
+public class CreateSongAction implements MenuAction {
+
+  private final CustomScanner scanner;
+  private final MenuActionContext ctx;
+  private final SongRepository repository;
+  private final AlbumRepository albumRepository;
+
+  @Override
+  public void execute() {
+    System.out.println("0) Przejdź do poprzedniego menu");
+
+    System.out.println("Podaj id albumu do którego dodajemy utwór: ");
+    var input1 = scanner.nextInt();
+    if (pressedZero(input1)) return;
+    var optAlbum = albumRepository.findById(input1);
+    Album album;
+    if (optAlbum.isPresent()) {
```

Comment:
to bym dał na koniec w metodzie pomocniczej któa by tak wygladała

```
private void addAlbum(SongBuilder builder) {
 System.out.println("Podaj id albumu do którego dodajemy utwór: ");
 int albumId = scanner.nextInt();
 albumRepository.findById(input1).ifPresentOrElse(album -> builder.album(album), () -> {
  System.out.println("Podany id albumi nie istnieje");
 addAlbum(builder);
 });
}
```

ewentualnie jeszczze dodac w tej metodzie ze jak poda 0 to wyjsc do głlownego menu